compiler-cli/src/publish.rs

```diff
@@ -754,3 +757,107 @@ fn prevent_publish_git_dependency() {
 fn quotes(x: &str) -> String {
     format!(r#"<<"{x}">>"#)
 }
+
+#[test]
+fn exported_project_files_test() {
+    let tmp = tempfile::tempdir().unwrap();
+    let path = Utf8PathBuf::from_path_buf(tmp.path().join("my_project")).expect("Non Utf8 Path");
+
+    let exported_project_files = &[
+        "LICENCE",
+        "LICENCE.md",
+        "LICENCE.txt",
+        "LICENSE",
+        "LICENSE.md",
+        "LICENSE.txt",
+        "NOTICE",
+        "NOTICE.md",
+        "NOTICE.txt",
+        "README",
+        "README.md",
+        "README.txt",
+        "gleam.toml",
+        "priv/wibble",
+        "priv/wobble.js",
+        "src/exported.gleam",
+        "src/exported_ffi.erl",
+        "src/exported_ffi.ex",
+        "src/exported_ffi.hrl",
+        "src/exported_ffi.js",
+        "src/exported_ffi.mjs",
+        "src/exported_ffi.ts",
+        "src/nested/exported.gleam",
+        "src/nested/exported_ffi.erl",
+        "src/nested/exported_ffi.ex",
+        "src/nested/exported_ffi.hrl",
+        "src/nested/exported_ffi.js",
+        "src/nested/exported_ffi.mjs",
+        "src/nested/exported_ffi.ts",
```

Comment:
This is missing the gitignored files!

Reply:
yes, they are in a separate list so that I don't have to specify them twice while building the `.gitignore`  file for the test. They are still unexported.

Reply:
I've made the test clearer with your feedback.

Reply:
Oh! I'm understanding the confusion now! There's a bug in the old code where it is excluding gitignored files from the publish for some reason. This shouldn't be the case, all files should be included.

Reply:
Done!